articles/10-tips-for-working-with-node.js/index.md

```diff
@@ -0,0 +1,295 @@
+
+-   [Browserify](https://github.com/browserify/browserify): analyses `require ()` calls in your Node.js application to create a bundle that you can serve the browser in a single `<script>` tag.
+
+-   [Nodemailer](https://nodemailer.com/about/): Allows (easy as cake) email sending.
+
+You can have a look at more available libraries on [GitHub](https://github.com/aravindnc/A-to-Z-List-of-Useful-Node.js-Modules).
+
+## 3. The Node.js HTTP Server Module
+
+Node.js. has a built-in module called HTTP (Hypertext Transfer Protocol). It helps in making requests as a client and running the server to accept requests and return responses. 
+
+In this case, when you visit a webpage from a browser, you make a request to another computer (webserver) via the internet, which then provides you the webpage response. The webserver receives HTTP requests from the client, which
+listens to a server port and provides an HTTP response for example a HTML Page. 
+
+To use HTTP and client server in Node.js you must use `require ('http');`. Let's look at a simple Hello World HTTP Server module that will listen on port number 3000.
+```js
+var http = require('http')
+
+var server = http.createServer((function(request, response) {
+
+    response.writeHead(200, { "Content-Type\": "text/plain" });
+
+    response.end("Hello World\n");
+
+}));
+server.listen(3000);
+```
+When you type localhost:3000 on your browser, The output will be "Hello
+World".
+
+Check out this [article](https://www.section.io/engineering-education/http-requests-nodejs/) to find out more about on working with HTTP and Node.js.
+
+## 4. Embracing Asynchronous Node.js Functions
+
+Asynchronous is heavily used in Node.js to ensure non-blocking operations flow. It enables a system to handle thousands of concurrent requests and process gigabytes of data with a small amount of RAM. 
+
+Asynchronous I/O permits other processing to continue even before the first transmission has finished. Asynchronous helps to avoid the so-called "Call-back Hell"
+
+Example of async/await in Node.js:
+```js
+async function myFunction(inputValue) {
+
+    try {
+
+        const a = await asyncFunc1('value');
+
+        const b = await asyncFunc2(a);
+
+        const c = syncFunc3(b);
+
+        return await asyncFunc4(c);
+
+    } catch (ex) {
+
+        *// handle exception*
+
+    }
+
+}
+```
+In this scenario, we can see that functions 1, 2 and 4 are
+asynchronous.
+
+## 5. Handling Node.js Errors
+
+The common Node.js errors are:
+
+![Node.js_error_types](/engineering-education/10-tips-for-working-with-node.js/node.js_error_types.png)
+
+In Node.js, errors are handled through exceptions. For example, if we
+have an error that would occur when we divide a number by zero, our
+Node.js application will crash, so then we need to handle the error to
+continue with the normal execution of our application i.e.
+```js
+try {
+
+    var m = 1;
+
+    var n = 1 / 0;
+
+} catch (err) {
+
+    *//Handle the error here.*
```

Comment:
Again, the asterisks need to be inside the slashes for it to be commented correctly.